src/main/java/com/google/devtools/build/lib/collect/nestedset/NestedSetFingerprintCache.java

```diff
@@ -60,6 +61,27 @@ public <T> void addNestedSetToFingerprint(
     addToFingerprint(mapFn, fingerprint, digestMap, children);
   }
 
+  public static <T> String describedNestedSetFingerprint(
+      MapFn<? super T> mapFn, NestedSet<T> nestedSet) {
+    if (nestedSet.isEmpty()) {
+      return "<empty>";
+    }
+    StringBuilder sb = new StringBuilder();
+    sb.append("order: ").append(nestedSet.getOrder()).append('\n');
+    List<T> list = nestedSet.toList();
+    sb.append("size: ").append(list.size()).append('\n');
+    for (T item : list) {
+      sb.append("  ");
+      try {
+        mapFn.expandToCommandLine(item, s -> sb.append(sb).append(", "));
+      } catch (CommandLineExpansionException | InterruptedException e) {
```

Comment:
don't swallow `InterruptedException`, that potentially makes us swallow Ctrl-C (or other events that require Bazel to stop what it's doing quickly)